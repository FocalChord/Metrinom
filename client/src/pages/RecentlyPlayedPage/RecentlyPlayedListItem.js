import { Avatar, Box, Divider, ListItem, ListItemAvatar, ListItemSecondaryAction, ListItemText } from "@material-ui/core";
import React from "react";
import { useHistory } from "react-router-dom";

const RecentlyPlayedListItem = ({ track }) => {
    const history = useHistory();

    return (
        <Box>
            <ListItem onClick={() => history.push(`/track/${track.trackId}`)} button>
                <ListItemAvatar>
                    <Avatar
                        style={{
                            borderRadius: 0,
                            width: 65,
                            height: 65,
                        }}
                        src={track.albumArt}
                    />
                </ListItemAvatar>
                <ListItemText
                    style={{
                        paddingLeft: 20,
                    }}
                    primary={track.trackName}
                    secondary={
                        <React.Fragment>
                            {track.artistName}
                            <br />
                            {track.albumName}
                        </React.Fragment>
                    }
                />
                <ListItemSecondaryAction>{track.dateString}</ListItemSecondaryAction>
            </ListItem>
            <Divider variant="inset" />
        </Box>
    );
};

export default RecentlyPlayedListItem;
