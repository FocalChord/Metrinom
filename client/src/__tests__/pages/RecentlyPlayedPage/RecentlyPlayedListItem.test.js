import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import RecentlyPlayedListItem from "../../../pages/RecentlyPlayedPage/RecentlyPlayedListItem";
import { MemoryRouter } from "react-router-dom";

test("<RecentlyPlayedListItem />", () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <RecentlyPlayedListItem
                    track={{
                        trackId: "321",
                        trackName: "Godzilla",
                        albumArt: "art",
                        albumName: "Music to be Murdered by",
                        dateString: "2020",
                    }}
                />
            </MemoryRouter>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
