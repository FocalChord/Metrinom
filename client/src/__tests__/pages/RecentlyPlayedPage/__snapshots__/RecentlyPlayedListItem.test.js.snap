// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentlyPlayedListItem /> 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <li
    className="MuiListItem-container"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemAvatar-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle"
          style={
            Object {
              "borderRadius": 0,
              "height": 65,
              "width": 65,
            }
          }
        >
          <img
            className="MuiAvatar-img"
            src="art"
          />
        </div>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-multiline"
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Godzilla
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <br />
          Music to be Murdered by
        </p>
      </div>
    </div>
    <div
      className="MuiListItemSecondaryAction-root"
    >
      2020
    </div>
  </li>
  <hr
    className="MuiDivider-root MuiDivider-inset"
  />
</div>
`;
